src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,58 @@
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Extensions;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+    private readonly ILogger<AddInsurance> _logger;
+
+    public AddInsurance(DataContext context, ILogger<AddInsurance> logger)
+    {
+        _context = context;
+        _logger = logger;   
+    }
+
+    public async Task Handle(InsuranceData item, HttpRequestData req)
+    {
+
+        if (string.IsNullOrEmpty(item.Pid))
+        {
+            _logger.LogWarning("Error, received pid is empty!");
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+        }
+
+        var sub = req.ReadPidFromJwt(); //secured 
+        if (item.Pid != sub)
+        {
+            _logger.LogWarning("Error, unauthorized access pid does not match with header");
+            throw new ApiException(ErrorCode.Unauthorized);
+        }
+
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Pid == item.Pid);
+        if (user is null)
+        {
+            _logger.LogWarning("Error reading user, user is null!");
+            throw new ApiException(ErrorCode.NotFound);
+        }
+
+        user.CarInsurance.CarBarnd = item.CarInsurance.CarBarnd;
+        user.CarInsurance.Year = item.CarInsurance.Year;
```

Comment:
You probably should check if the year is valid. Like something less that 1900 is not valid.

Reply:
This goes about other dates